src/main/scala/com/astrolabsoftware/grafink/processor/VertexProcessor.scala

```diff
@@ -0,0 +1,156 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.astrolabsoftware.grafink.processor
+
+import org.apache.spark.sql.{ DataFrame, Row, SaveMode, SparkExtensions, SparkSession }
+import org.apache.spark.sql.types.DataType
+import org.apache.tinkerpop.gremlin.structure.T
+import org.janusgraph.graphdb.database.StandardJanusGraph
+import zio.{ Has, URLayer, ZIO, ZLayer }
+import zio.blocking.Blocking
+import zio.logging.Logging
+
+import com.astrolabsoftware.grafink.JanusGraphEnv
+import com.astrolabsoftware.grafink.JanusGraphEnv.JanusGraphEnv
+import com.astrolabsoftware.grafink.Job.{ JobTime, SparkEnv }
+import com.astrolabsoftware.grafink.common.{ PartitionManager, Utils }
+import com.astrolabsoftware.grafink.models.JanusGraphConfig
+import com.astrolabsoftware.grafink.models.config.Config
+import com.astrolabsoftware.grafink.services.IDManager.{ IDManagerService, IDType }
+
+object VertexProcessor {
+
+  val vertexDir = "vertex"
+
+  type VertexProcessorService = Has[VertexProcessor.Service]
+
+  trait Service {
+    def processData(jobTime: JobTime, df: DataFrame): ZIO[IDManagerService with Logging, Throwable, DataFrame]
+    def process(jobTime: JobTime, df: DataFrame): ZIO[IDManagerService with Logging, Throwable, Unit]
+    def load(df: DataFrame): ZIO[Logging, Throwable, Unit]
+  }
+
+  val live: URLayer[SparkEnv with Has[JanusGraphConfig] with Logging, VertexProcessorService] =
+    ZLayer.fromEffect(
+      for {
+        spark            <- ZIO.access[SparkEnv](_.get.sparkEnv)
+        janusGraphConfig <- Config.janusGraphConfig
+      } yield new VertexProcessorLive(spark, janusGraphConfig)
+    )
+
+  def processData(
+    jobTime: JobTime,
+    df: DataFrame
+  ): ZIO[VertexProcessorService with IDManagerService with Logging, Throwable, DataFrame] =
+    ZIO.accessM(_.get.processData(jobTime, df))
+
+  def process(
+    jobTime: JobTime,
+    df: DataFrame
+  ): ZIO[VertexProcessorService with IDManagerService with Logging, Throwable, Unit] =
+    ZIO.accessM(_.get.process(jobTime, df))
+
+  def load(df: DataFrame): ZIO[VertexProcessorService with Logging, Throwable, Unit] =
+    ZIO.accessM(_.get.load(df))
+
+}
+
+final class VertexProcessorLive(spark: SparkSession, config: JanusGraphConfig) extends VertexProcessor.Service {
+
+  def addId(df: DataFrame, lastMax: IDType): DataFrame =
+    SparkExtensions.zipWithIndex(df, lastMax + 1)
+
+  def processData(jobTime: JobTime, df: DataFrame): ZIO[IDManagerService with Logging, Throwable, DataFrame] = {
+
+    val mode = SaveMode.Append
+
+    for {
```

Comment:
So If I understand correctly, the job:
1. read original Fink data on HDFS
2. select columns of interests, and rename columns 
3. add IDs
4. write data on HDFS (parquet)
5. reload data, and push to JanusGraph

Why this intermediary step (4) to write the data on HDFS?

Reply:
This is to read it again when adding the edges, so we get the ids and from ids we can get vertex objects, hence we do not have to query janusgraph which will definitely be slower

Reply:
OK I see. Sounds good to me.